default.nix

```diff
@@ -0,0 +1,154 @@
+{ pkgs ? import <nixpkgs> {} }:
+with rec {
+  loadNixpkgs = version:
+    let
+      src = builtins.fetchTarball "https://nixos.org/channels/${version}/nixexprs.tar.xz";
+    in import src {};
+
+  nixosFromNixpkgs = nixpkgs:
+    import "${nixpkgs.path}/nixos";
+
+  nixpkgsdocsFromNixpkgs = nixpkgs:
+    import "${nixpkgs.path}/doc";
+
+  latestJobsetFromHydra = { project, jobset, job }:
+    let
+      url = "https://hydra.nixos.org/job/${project}/${jobset}/${job}/latest.json";
+      json = builtins.fetchurl { inherit url; };
+      builddata = builtins.fromJSON (builtins.trace json json);
+      path = builtins.path {
+        path = builddata.buildoutputs.out.path;
+      };
+    in null;
+
+  projectFromGitHub = { repo, branch, args, file ? "default.nix" }:
+    let
+      src = builtins.fetchGit {
+        url = "https://github.com/nixos/${repo}.git";
+        ref = branch;
+      };
+    in pkgs.callPackage "${src}/${file}" args;
+
+  nixos-18-03 = loadNixpkgs "nixos-18.03";
+  nixos-unstable = loadNixpkgs "nixos-unstable";
+
+
+  callNix = name: command:
+    pkgs.runCommand name rec {
+      buildInputs = with pkgs; [ nix ];
+      NIX_STORE_DIR = "/tmp/test-root/store";
+      NIX_LOCALSTATE_DIR = "/tmp/test-root/var";
+      NIX_LOG_DIR = "/tmp/test-root/var/log/nix";
+      NIX_STATE_DIR = "/tmp/test-root/var/nix";
+      NIX_CONF_DIR = "/tmp/test-root/etc";
+    } command;
+
+  package-list-stable = callNix "package-list-stable.nix"
+    ''
+      nix-env --readonly-mode -f '<nixpkgs>' -I nixpkgs=${nixos-18-03.path} -qa --json --arg config 'import ${./discover-more-packages-config.nix}' > $out
+    '';
+
+  package-list-unstable = callNix "package-list-unstable.nix"
+    ''
+      nix-env --readonly-mode -f '<nixpkgs>' -I nixpkgs=${nixos-unstable.path} -qa --json --arg config 'import ${./discover-more-packages-config.nix}' > $out
+    '';
+};
+pkgs.runCommand "nixos.org" rec {
+  src = ./.;
+
+  buildInputs = with pkgs; [
+    perl
+    python
+    perlPackages.TemplateToolkit
+    perlPackages.TemplatePluginJSONEscape
+    perlPackages.TemplatePluginIOAll
+    perlPackages.XMLSimple
+    libxslt libxml2 imagemagick git curl
+    xhtml1
+    nixStable
+    gnupg
+    jq
+  ];
+
+  stable_packages = package-list-stable;
+  unstable_packages = package-list-unstable;
+  stable_options = (import "${nixos-18-03.path}/nixos/release.nix" {}).options;
+
+  nix_pills = projectFromGitHub {
+    repo = "nix-pills";
+    branch = "master";
+    args = {
+      revCount = 0;
+      shortRev = "abc123";
+    };
+  };
+
+  hydra = let
+      src = builtins.fetchGit {
+        url = "https://github.com/nixos/hydra.git";
+        ref = "master";
+      };
+      imported = import "${src}/release.nix" {
+        hydraSrc = src;
+      };
+    in imported.manual;
```

Comment:
the above two aren't great. These two, and the other impure parts, could be made as a pre-nix-build step.